src/Symfony/Installer/NewCommand.php

```diff
@@ -415,13 +415,19 @@ protected function generateComposerProjectName()
      */
     private function fixComposerPackageName($name)
     {
-        return strtolower(
-            preg_replace(
-                array('/([A-Z]+)([A-Z][a-z])/', '/([a-z\d])([A-Z])/'),
-                array('\\1-\\2', '\\1-\\2'),
-                strtr($name, '-', '.')
-            )
+        $name = preg_replace(
+            array('/([A-Z]+)([A-Z][a-z])/u', '/([a-z\d])([A-Z])/u'),
+            array('\\1-\\2', '\\1-\\2'),
+            strtr($name, '-', '.')
         );
+
+        if (function_exists('mb_strtolower')) {
+            $name = mb_strtolower($name);
+        } else {
+            $name = utf8_encode(strtolower(utf8_decode($name)));
```

Comment:
Just another thought... don't know how USERNAME can look for non latin countries (i.e. Kanji ?), perhaps it would be safer to replace chars by underscores?

Or fix vendor/name separately (and ensure it is not empty)?

    protected function generateComposerProjectName()
    {
        $name = $this->projectName;
        
        if (!empty($_SERVER['USERNAME'])) {
            $vendor = $_SERVER['USERNAME'];
        } elseif (true === extension_loaded('posix') && $user = posix_getpwuid(posix_getuid())) {
            $vendor = $user['name'];
        } elseif (get_current_user()) {
            $vendor = get_current_user();
        } else {
            $vendor = $name;
        }
        $vendor = $this->fixComposerPackageName($name);
        $name = $this->fixComposerPackageName($name);
        
        // package names must be in the format foo/bar
        return $vendor.'/'.$name;
    }